models/Schematic_DataModel.php

```diff
@@ -40,14 +40,21 @@ protected function defineAttributes()
      * Populate data model from yaml.
      *
      * @param string $yaml
+     * @param string $overrideYaml
      *
      * @return Schematic_DataModel
      */
-    public static function fromYaml($yaml)
+    public static function fromYaml($yaml, $overrideYaml)
     {
         $data = Yaml::parse($yaml);
+        $overrideData = Yaml::parse($overrideYaml);
+        if ($overrideData != null) {
+            $mergedData = array_replace_recursive($data, $overrideData);
```

Comment:
Can you explain the difference? Because I think `array_replace_recursive` does exactly what we want here, and my tests indicate it works correctly. According to the documentation `\CMap::mergeArray` does exactly the same.